src/Module.purs

```diff
@@ -42,13 +41,13 @@ import Util.Set ((∪))
 parse :: forall a m. MonadError Error m => String -> SParser a -> m a
 parse src = liftEither <<< lmap (E.error <<< show) <<< runParser src
 
-parseProgram :: forall m. LoadFile m => Array Folder -> File -> AffError m (Raw S.Expr × List ModuleName)
-parseProgram folders (File file) =
-   loadFile folders (File (file <> fluidExtension)) >>= flip parse P.program
+parseProgram :: forall m. String -> AffError m (Raw S.Expr × List ModuleName)
+parseProgram fluidSrc = flip parse P.program fluidSrc
 
 datasetAs :: forall m. MonadAff m => MonadError Error m => LoadFile m => Array Folder -> Bind File -> Raw ProgCxt -> m (Raw ProgCxt)
 datasetAs folders (x ↦ file) (ProgCxt r@{ datasets }) = do
-   eα <- (fst <$> parseProgram folders file) >>= desug
+   s × _ <- (\src -> parseProgram src) =<< loadFile folders file
```

Comment:
The `loadFile` bit was fine as it was -- the bit I was suggesting you could keep as it was is the `parseProgram`/`desug` logic, which I don't think is affected.